### PR TITLE
Added PHPCS config and dependencies

### PR DESCRIPTION
**Pull request by seebeen**
Implemented PSR-12 PHPCS config and added .editorconfig for development standardization

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,9 @@
     "require-dev": {
         "phpunit/phpunit": "^9",
         "mockery/mockery": "^1.6",
-        "phpstan/phpstan": "^1"
+        "phpstan/phpstan": "^1",
+        "dealerdirect/phpcodesniffer-composer-installer": "^1.0",
```
Comment (olivervogel) [inline composer.json:23]: Why do we need this package?

File: phpcs.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<ruleset name="Intervention/image">
+
+  <rule ref="PSR12"/>
+
+  <rule ref="Generic.Files.LineLength">
+    <severity>0</severity>
+  </rule>
+
+  <file>src/</file>
```
Comment (olivervogel) [inline phpcs.xml:10]: Folder `tests` should also be included.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# editorconfig.org
```
Comment (olivervogel) [inline .editorconfig:1]: `.editorconfig` should be excluded via `.gitattributes`

File: composer.json (hunk 1 above)
Comment (seebeen) [reply composer.json:23]: To automatically setup phpcs for various IDEs - so our phpcs.xml config works OOB

File: phpcs.xml (hunk 2 above)
Comment (seebeen) [reply phpcs.xml:10]: Roger roger

File: .editorconfig (hunk 3 above)
Comment (seebeen) [reply .editorconfig:1]: Done

**Comment by seebeen** [comment]
Changes implemented per request.

File: composer.json (hunk 1 above)
Comment (olivervogel) [reply composer.json:23]: I tried to figure out what this package does but still don't get it.

> This composer installer plugin allows for easy installation of PHP_CodeSniffer
> No more symbolic linking of directories, checking out repositories on specific locations or changing the phpcs configuration.

Why do I need an installation package for "PHP_CodeSniffer" when you can install the tool without any other dependencies? `phpcs.xml` is read in automatically and/or can be integrated via the GitHub testing workflow.

Don't get me wrong. I don't want to criticize here, but always try to keep package dependencies as low as possible.

I think the IDE setup belongs on the machine of the developer. In my opinion, we can do without this dependency in this case.

Comment (seebeen) [reply composer.json:23]: OK, I'll remove it.

**Comment by seebeen** [comment]
And again 😅

**Comment by olivervogel** [comment]
Thank you very much. I included the code standard validation additionally in the Github Workflow and added validation of line length maximum. Had to tweak 3 lines for the checks to pass.

**Comment by seebeen** [comment]
Awesome. Continuing with the Trim implementation.

**Pull request merged**